src/components/NewCommentForm.tsx

```diff
@@ -1,71 +1,64 @@
 import classNames from 'classnames';
-import React, { useState } from 'react';
+import React from 'react';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { addComment } from '../features/comments';
+import {
+  clearAllData,
+  selectFormData,
+  selectFormErrors,
+  setAllErrors,
+  setValue,
+} from '../features/newCommentForm';
 import { CommentData } from '../types/Comment';
 
-type Props = {
-  onSubmit: (data: CommentData) => Promise<void>;
-};
-
-export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
-  const [submitting, setSubmitting] = useState(false);
-
-  const [errors, setErrors] = useState({
-    name: false,
-    email: false,
-    body: false,
-  });
+type Props = {};
 
-  const [{ name, email, body }, setValues] = useState({
-    name: '',
-    email: '',
-    body: '',
-  });
+export const NewCommentForm: React.FC<Props> = ({}) => {
+  const dispatch = useAppDispatch();
+  const selectedPost = useAppSelector(state => state.selectedPost);
+  const { isSubmiting } = useAppSelector(state => state.comments);
+  const { name, email, body } = useAppSelector(selectFormData);
+  const errors = useAppSelector(selectFormErrors);
 
   const clearForm = () => {
-    setValues({
-      name: '',
-      email: '',
-      body: '',
-    });
-
-    setErrors({
-      name: false,
-      email: false,
-      body: false,
-    });
+    dispatch(clearAllData());
   };
 
   const handleChange = (
     event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
   ) => {
-    const { name: field, value } = event.target;
+    const { name: fieldName, value } = event.target;
 
-    setValues(current => ({ ...current, [field]: value }));
-    setErrors(current => ({ ...current, [field]: false }));
+    const field = fieldName as keyof CommentData;
+
+    dispatch(setValue({ field, value }));
   };
 
   const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
-    setErrors({
-      name: !name,
-      email: !email,
-      body: !body,
-    });
+    dispatch(
+      setAllErrors({
+        name: !name,
+        email: !email,
+        body: !body,
+      }),
+    );
 
     if (!name || !email || !body) {
       return;
     }
 
-    setSubmitting(true);
-
-    // it is very easy to forget about `await` keyword
-    await onSubmit({ name, email, body });
+    dispatch(
+      addComment({
+        name: name,
+        email: email,
+        body: body,
+        postId: selectedPost!.id,
+      }),
+    );
 
-    // and the spinner will disappear immediately
-    setSubmitting(false);
-    setValues(current => ({ ...current, body: '' }));
-    // We keep the entered name and email
+    dispatch(setValue({ field: 'body', value: '' }));
   };
```

Comment:
I would keep state of from locally. You don't have to keep all the state inside store. You can still use useState or useReducer.